scripts/research/lib/batch02_contracts.py

```
@@ -368,38 +439,106 @@ def rolling_midrank_percentile(
+        raise Batch02ContractError("lookback_ms must be a positive integer")
+    try:
+        times = np.asarray(timestamps_ms)
+    except (TypeError, ValueError) as exc:
+        raise Batch02ContractError("timestamps_ms must be one-dimensional integers") from exc
+    if times.ndim != 1 or len(times) != len(values) or times.dtype.kind not in "iu":
+        raise Batch02ContractError(
+            "timestamps_ms must be a one-dimensional integer array matching series"
+        )
+    times = times.astype(np.int64)
+    if len(times) > 1 and np.any(times[1:] <= times[:-1]):
+        raise Batch02ContractError("timestamps_ms must be strictly increasing")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)